internal/service/domain/sensor/sensor.go

```diff
@@ -100,6 +104,30 @@ func (s *SensorService) Delete(ctx context.Context, id string) error {
 	return nil
 }
 
+func (s *SensorService) MapSensorToTree(ctx context.Context, sen *entities.Sensor) error {
+	if sen == nil {
+		return errors.New("sensor cannot be nil")
+	}
+
+	if err := s.validator.Struct(sen); err != nil {
+		return service.NewError(service.BadRequest, errors.Wrap(err, "validation error").Error())
+	}
```

Comment:
OK, but what do you have to validate if the entity has no annotations?

Reply:
And in this case, a sensor should already be defined, existing and validated, or not?